xapian-letor/ranker/listmle_ranker.cc

```diff
@@ -0,0 +1,211 @@
+/** @file listmle_ranker.cc
+ *  @brief Implementation of ListMLERanker
+ */
+/* Copyright (C) 2014 Hanxiao Sun
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+/*
+ListMLE is adapted from the paper:
+Fen Xia, Tie-Yan Liu, Jue Wang, Wensheng Zhang, Hang Li
+et al. "Listwise Approach to Learning to Rank - Theory and Algorithm."
+*/
+#include <config.h>
```

Comment:
The citation should be in the same comment, or else it will just be a comment rather than part of the documentation.

For an example of a similar situation, see xapian-core/api/editdistance.cc which appears in the collated source documentation as:

https://xapian.org/docs/sourcedoc/html/editdistance_8cc.html#details